halo2_common/src/arithmetic.rs

```diff
@@ -25,6 +25,10 @@ where
 {
 }
 
+#[deprecated(
+    since = "0.3.2",
+    note = "please use ZAL api engine instead,\nsee: https://github.com/privacy-scaling-explorations/halo2/issues/216"
+)]
 fn multiexp_serial<C: CurveAffine>(coeffs: &[C::Scalar], bases: &[C], acc: &mut C::Curve) {
```

Comment:
We just realized that this function is used in the halo2 `best_multiexp` implementation, but we already have that implemented in `halo2curves` (which is the one `H2cEngine` is using).  We should probably delete the msm implementation in halo2 and use the one from `halo2curves` instead via the `H2cEngine`, or via Zal (to support any msm implementation).